### DISCUSSION TITLE
Need API to programmatically close child window after form submission

### DISCUSSION THREAD

**Discussion by sadlrxn** [slint-ui/slint#6640] [Q&A]
### Description
I'm working on a Slint UI application where my main window contains a button that opens a new child window with a form. I want to automatically close this child window after the form is successfully submitted (when no validation errors are found).

Currently, window closing is only handled through the `on_close_request` event, which is triggered by window management actions (like clicking the X button). However, I need a way to programmatically close the window from within my code after form submission.

### Current Behavior
- Main window opens child window via button click
- Form submission is processed
- Window remains open even after successful submission
- Window can only be closed through window manager actions

### Desired Behavior
- Main window opens child window via button click
- Form submission is processed
- If submission is successful (no errors), window automatically closes
- If submission has errors, window remains open

### Question
Is there an API method available to programmatically close a Slint window? Looking for something similar to:
```rust
// Pseudo-code of what I'm looking for
if form_submission.is_successful() {
    child_window.close();
}
```
or 

- The `on_close_request` event handler is not sufficient for this use case as it only handles window manager close events
- Need a programmatic way to trigger window closure from within the application logic

similar issue:   #5465 marked answered but it's not.

**Comment by Enyium** [comment]
Why to you post this as a discussion *and* [as an issue](https://github.com/slint-ui/slint/issues/6641)?!

**Reply by sadlrxn** [reply]
My bad, it was a mistake. Also, droping a strong reference doesn't close the window.
Using .hide() doesn't free up the memory it is using.  or maybe i am doing it wrong.

```rust

            update_window
                .unwrap()
                .global::<TableData>()
                .on_send_close_request({
                    let weak_app = Rc::clone(&update_window);
                    move || {
                        println!("droping update window");
                        drop(weak_app.upgrade().unwrap());
                    }
                });


```

**Reply by Enyium** [reply]
You're dropping a strong reference you just created via `upgrade()`. That's like `i += 1; i -= 1;`. I think you only need to manage your own strong references and their dropping, if you're holding them in a long-living struct. If you let Slint manage your window component references and aren't holding on to strong references anywhere else, `hide()` should suffice. EDIT: E.g., if you're instatiating a window component, `show()`ing it, and distributing it in the form of weak references (via `.as_weak()`) moved into event handler closures, Slint's internals is the only place holding a strong reference to your window component, and `hide()` drops this reference.

**Comment by sadlrxn** [comment]
A
